### PR TITLE
NEXTEUROPA-5898 : Fix coding standards in multisite_path_override,

### PR DESCRIPTION
**Pull request by senseca**
multisite_twitterblock, multisite_readspeaker

### CODE REVIEW

File: phpcs-ruleset.xml [hunk 1]
```diff
@@ -12,6 +12,7 @@
     <!-- Minified files don't have to comply with coding standards. -->
     <exclude-pattern>*.min.css</exclude-pattern>
     <exclude-pattern>*.min.js</exclude-pattern>
+    <exclude-pattern>profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/</exclude-pattern>
```
Comment (pfrenssen) [inline phpcs-ruleset.xml]: Oops! These seem to be proprietary files owned by ReadSpeaker, and I'm not sure whether we are allowed to redistribute them. I will open a new ticket for this.

In scope of this ticket, this exclude pattern is excluding everything in that folder, not only the minified javascript files, please make it more specific by appending `*.js` to the end of the path.

File: phpcs-ruleset.xml [hunk 2]
```diff
@@ -78,6 +76,7 @@
     <!-- Views handlers not strictly follow Drupal function name conventions. -->
     <rule ref="Drupal.NamingConventions.ValidFunctionName">
         <exclude-pattern>profiles/common/modules/custom/ecas/ecas_extra/includes/views/handlers/*.inc</exclude-pattern>
+        <exclude-pattern>profiles/common/modules/features/multisite_twitterblock/plugins/*.inc</exclude-pattern>
```
Comment (pfrenssen) [inline phpcs-ruleset.xml]: This is not a Views handler, but a Context reaction. It's fine to update the documentation to say that both Views handlers and Context reactions do not follow the conventions.

There is only one file affected, so don't use a wildcard here. The plugins/ folder is generic. If more plugins are added in the future we want to be sure they are scanned for this coding standard.

File: profiles/common/modules/features/multisite_path_override/multisite_path_override.module [hunk 3]
```diff
@@ -2,49 +2,40 @@
...
 
-/*
+/**
  * Implements hook_pathauto_alias_alter().
  */
 function multisite_path_override_pathauto_alias_alter(&$alias, array &$context) {
   // Replace punctuation with string value.
-
-  $alias = str_replace("-","moins", $alias);
+  $alias = str_replace("-", "moins", $alias);
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_path_override/multisite_path_override.module]: ?!?? Out of scope for this ticket, but this is very strange. This cannot ever habe been the intention.

File: profiles/common/modules/features/multisite_readspeaker/tests/multisite_readspeaker.test [hunk 4]
```diff
@@ -17,13 +24,19 @@ class ReadSpeakerTestCase extends DrupalWebTestCase {
...
-    $this->default_user = $this->drupalCreateUser(array('display multisite_readspeaker'));
+    $this->adminUser = $this->drupalCreateUser(array('administer blocks', 'administer multisite_readspeaker'));
+    $this->defaultUser = $this->drupalCreateUser(array('display multisite_readspeaker'));
   }
 
+  /**
+   * Tests permissions.
+   */
   public function testPermission() {
     $data = module_invoke('multisite_readspeaker', 'permission');
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_readspeaker/tests/multisite_readspeaker.test]: This test is completely pointless, just remove it.

File: profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module [hunk 5]
```diff
@@ -199,15 +198,16 @@ function multisite_twitterblock_theme() {
 
 /**
  * Implements theme hook.
- * @global bool $is_https
  *
  * @param array $variables
- *   array of new variables
+ *   Array of new variables.
+ *
+ * @global bool $is_https
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module]: Remove the `@global bool` line, globals are not documented like this in Drupal.

File: profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module [hunk 6]
```diff
@@ -199,15 +198,16 @@ function multisite_twitterblock_theme() {
...
  *
  * @return string
- *   twitter output config
+ *   Twitter output config.
  */
-function theme_fpfis_twitter($variables) {
+function theme_fpfis_twitter(array $variables = array()) {
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module]: Don't add ` = array()` unless the array is optional. The code expects that there is a `$variables['type']` set, so it is not optional in this case.

File: profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module [hunk 7]
```diff
@@ -402,11 +402,11 @@ function multisite_twitterblock_form_node_form_alter(&$form, $form_state) {
 
 /**
  * Provides a settings subform.
- * 
+ *
  * @param array $form
  *   Rendable array.
  */
-function _multisite_twitterblock_entity_subform(&$form, $fpfis_twitter_override = array()) {
+function _multisite_twitterblock_entity_subform(array &$form = array(), $fpfis_twitter_override = array()) {
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module]: The `$form` array is not optional, so don't append ` = array()` to it. Prefixing it with `array ` is a very good practice though.

File: profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module [hunk 8]
```diff
@@ -483,12 +483,12 @@ function multisite_twitterblock_check_availability($value, $variable_name) {
  * Loads overriding config from db.
  *
  * @param int $entity_id
- *   Entity Id
+ *   Entity Id.
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module]: Id must be spelled in capital letters (ID) according to the coding standards.

File: profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module [hunk 9]
```diff
@@ -933,13 +934,13 @@ function multisite_twitterblock_get_var_name($machine_name, $entity_type) {
...
  * Submit handler for Node type form.
- * 
+ *
  * @param array $form
  *   Form array.
  * @param array $form_state
  *   Form state array.
  */
-function multisite_twitterblock_form_node_type_submit(&$form, $form_state) {
+function multisite_twitterblock_form_node_type_submit(array &$form = array(), array $form_state = array()) {
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module]: `$form` and `$form_state` are not optional, so do not append ` = array()`.

File: profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module [hunk 10]
```diff
@@ -969,13 +970,13 @@ function multisite_twitterblock_form_taxonomy_form_vocabulary_alter(&$form, $for
...
  * Submit handler for taxonomy vocabulary form.
- * 
+ *
  * @param array $form
  *   Form array.
  * @param array $form_state
  *   Form state array.
  */
-function multisite_twitterblock_form_taxonomy_vocabulary_submit(&$form, $form_state) {
+function multisite_twitterblock_form_taxonomy_vocabulary_submit(array &$form = array(), array $form_state = array()) {
```
Comment (pfrenssen) [inline profiles/common/modules/features/multisite_twitterblock/multisite_twitterblock.module]: `$form` and `$form_state` are not optional, so do not append ` = array()`.

File: phpcs-ruleset.xml (hunk 1 above)
Comment (senseca) [inline phpcs-ruleset.xml]: Exactly. I wanted to load the library automatically in makefile. But we have to be logged to download them. It's a good thing to create a new ticket for this.

**Pull request merged**